File: app/src/main/java/com/hana053/micropost/interactors/MicropostInteractor.java
package com.hana053.micropost.interactors;

import com.hana053.micropost.domain.Micropost;

import retrofit2.http.Body;
import retrofit2.http.POST;
import rx.Observable;

public interface MicropostInteractor {

    @POST("microposts")
    Observable<Micropost> create(@Body Micropost micropost);
}

File: app/src/main/java/com/hana053/micropost/interactors/MicropostInteractor.kt
package com.hana053.micropost.interactors

import com.hana053.micropost.domain.Micropost
import retrofit2.http.Body
import retrofit2.http.POST
import rx.Observable


interface MicropostInteractor {

    @POST("microposts")
    fun create(@Body request: MicropostRequest): Observable<Micropost>

    data class MicropostRequest(val content: String)
}